service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -229,6 +234,18 @@ public DatasetPreviewTablesResponse listDatasetPreviewTables(
     return new DatasetPreviewTablesResponse().tables(tableMetadataList);
   }
 
+  private void validateSchema(String metaData) {
+    JsonNode node;
+    try {
+      node = objectMapper.readValue(metaData, JsonNode.class);
+    } catch (JsonProcessingException e) {
+      throw new IllegalMetadataException(e);
```

Comment:
Unlike the other code that's calling `objectMapper.readValue()` (in `DatasetService.toJsonNode()`), this exception could easily be seen by a user, so it should have a user visible message along with the caught exception.